### DISCUSSION TITLE
自動偵測是否有物體經過

### DISCUSSION THREAD

**Discussion by zhilin93517** [1101-GS4719-C/Python-Community-GS4719-C#2166] [General]
我曾經因為一些個人需求，需要偵測特定區域內是否有物體經過。最一開始的想法是使用機械式結構偵測，若有物體經過該區域，會觸發微動開關，微動開關會輸出訊號告訴我的電腦該如何做判別。這樣可以很有效率的對是否有物體經過這件事做出反應，達到我想要的需求。

**Comment by LeoTsai999** [comment]
請問你為什麼會製造這個裝置呢？我覺得這個裝置可能可以改變世界的運行法則，希望你可以開源它！

**Reply by zhilin93517** [reply]
這是個很有趣的問題，但如同我所說的，我因為個人需求所以做出了這種機械式結構進行偵測，我認為這樣的說法算是非常詳細了，希望這樣有回答到你的問題!

**Comment by zhilin93517** [comment]
後來我也有嘗試過使用TCRT5000模組進行偵測，因為在文中的方式雖然可行，但 flux efficiency 並不高，所以改使用此方式的話可以提高偵測的靈敏度與效率。

**Comment by zhilin93517** [comment]
TCRT5000的工作原理很簡單，從概念上論，他其實就是一個包裝在一起的紅外線發射器跟紅外線接收器，若紅外線接收器有接收到紅外線，則TCRT5000輸出訊號，這樣傳回到我的電腦，我就可以根據當前的 event 進行處理。

**Comment by zhilin93517** [comment]
若有一個物體經過TCRT5000，則TCRT5000發射出的紅外線可以被物體反射，再進入紅外線接收器中，這就是整個系統工作的原理。我認為這樣的設計非常巧妙，讓使用者可以親自設計並使用。

**Comment by seastar0687** [comment]
光閘也是不錯的選擇，可以使用光敏電阻跟led來實現，但可能要搭配arduino或是樹莓派來控制。
